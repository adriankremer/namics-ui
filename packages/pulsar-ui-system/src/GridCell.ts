import { cx, css } from "emotion";
import { GridCellProps } from "@pulsar-ui/core";
import { PulsarBoxProps } from "./Box";
import { breakpoints } from "./utils/systemVariables";

export type PulsarGridCellProps = PulsarBoxProps & {
  system: {
    cellWidth?: number[];
    cellHeight?: number[];
  };
};

export function useGridCellProps(
  { system }: PulsarGridCellProps,
  { className, ...htmlProps }: GridCellProps
) {
  const { cellWidth = [], cellHeight = [] } = system;
  const initialWidth = cellWidth.shift();
  const initialheight = cellHeight.shift();
  const gridCell = css`
    grid-column-end: span ${initialWidth || "12"};
    grid-row-end: span ${initialheight || "1"};
    align-items: center;
    ${cellWidth.map(
      (column, index) => `
      @media screen and (min-width: ${breakpoints[index]}) {
        grid-column-end: span ${column};
      }
    `
    )}
  `;
  return { ...htmlProps, className: cx(gridCell, className) };
}
