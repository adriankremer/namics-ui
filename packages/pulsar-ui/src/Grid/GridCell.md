```js
<Grid>
  <Grid.Cell width={6} height={4}>
    <Avatar width="100%" src="https://placekitten.com/400/400" />
  </Grid.Cell>
  <Grid.Cell width={3} height={1}>
    <Avatar width="100%" src="https://placekitten.com/300/300" />
  </Grid.Cell>
  <Grid.Cell width={1} height={1}>
    <Avatar width="100%" src="https://placekitten.com/150/150" />
  </Grid.Cell>
</Grid>
```
