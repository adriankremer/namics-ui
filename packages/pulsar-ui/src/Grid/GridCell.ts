import use from "reuse";
import { theme, prop, ifProp } from "styled-tools";
import Box, { BoxProps } from "../Box";
import styled from "../styled";

export interface CellProps extends BoxProps {
  sm?: number;
  md?: number;
  lg?: number;
  xl?: number;
  width?: number;
  height?: number;
}

const GridCell = styled(Box)<CellProps>`
  grid-column-end: span ${prop("width", "12")};
  grid-row-end: span ${prop("height", "1")};

  @media screen and (min-width: ${theme("breakpoints.sm", "768px")}) {
    grid-column-end: span ${ifProp("sm", prop("sm"), prop("width"))};
  }
  @media screen and (min-width: ${theme("breakpoints.md", "768px")}) {
    grid-column-end: span ${ifProp("md", prop("md"), prop("width"))};
  }
  @media screen and (min-width: ${theme("breakpoints.lg", "1024px")}) {
    grid-column-end: span ${ifProp("lg", prop("lg"), prop("width"))};
  }
  @media screen and (min-width: ${theme("breakpoints.xl", "1200px")}) {
    grid-column-end: span ${ifProp("xl", prop("xl"), prop("width"))};
  }
  ${theme("GridCell")}
`;

export default use(GridCell, "div");
