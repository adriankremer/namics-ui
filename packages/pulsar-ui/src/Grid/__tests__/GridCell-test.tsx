import * as React from "react";
import { render } from "react-testing-library";
import GridCell from "../GridCell";

test("html, attr", () => {
  const { getByTestId } = render(
    <GridCell id="grid" aria-label="test" data-testid="test" />
  );
  expect(getByTestId("test")).toHaveAttribute("id", "grid");
  expect(getByTestId("test")).toHaveAttribute("aria-label", "test");
});

test("styles", () => {
  const { container } = render(<GridCell />);
  expect(container.firstChild).toMatchSnapshot();
});
