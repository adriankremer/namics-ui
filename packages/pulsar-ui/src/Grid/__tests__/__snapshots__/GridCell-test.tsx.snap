// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styles 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  grid-column-end: span 12;
  grid-row-end: span 1;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

@media screen and (min-width:768px) {
  .emotion-0 {
    grid-column-end: span;
  }
}

@media screen and (min-width:768px) {
  .emotion-0 {
    grid-column-end: span;
  }
}

@media screen and (min-width:1024px) {
  .emotion-0 {
    grid-column-end: span;
  }
}

@media screen and (min-width:1200px) {
  .emotion-0 {
    grid-column-end: span;
  }
}

<div
  class="emotion-0"
/>
`;
